spackbot/pr_add_comments.py

```diff
@@ -39,17 +43,139 @@ def say_hello():
 
 - `@spackbot hello`: say hello and get a friendly response back!
 - `@spackbot help` or `@spackbot commands`: see this message 
+- `@spackbot fix style`: ask me to fix a failed style check
 
 I'll also help to label your pull request and assign reviewers!
 If you need help or see there might be an issue with me, open an issue [here](https://github.com/spack/spack-bot/issues)
 """
 
 
+def get_style_message(output):
+    """
+    Given a terminal output, wrap in a message
+    """
+    return (
+        """
+I was able to run `spack style --fix` for you!
+    
+```bash
+%s
+```    
+Keep in mind that I cannot fix your flake8 or mypy errors, so if you have any you'll need to fix them and update the pull request."""
+        % output
+    )
+
+
+def is_up_to_date(output):
+    """
+    A commit can fail if there are no changes!
+    """
+    return "branch is up to date" in output
+
+
+async def fix_style(event, gh, token):
+    """
+    Respond to a request to fix style.
+
+    We first retrieve metadata about the pull request. If the request comes
+    from anyone with write access to the repository, we commit, and we commit
+    under the identity of the original person that opened the PR.
+    """
+    response = requests.get(event.data["issue"]["pull_request"]["url"])
+    pr = response.json()
+
+    # Get the sender of the PR - do they have write?
+    sender = event.data["sender"]["login"]
+    repository = event.data["repository"]
+    collaborators_url = repository["collaborators_url"]
+
+    # If they don't have write, we don't allow the command
+    if not await found(gh.getitem(collaborators_url, {"collaborator": sender})):
+        logger.info(f"Not found: {sender}")
+        return (
+            "Sorry %s, I cannot do that for you. Only users with write can make this request!"
+            % sender
+        )
+
+    # Tell the user the style fix is going to take a minute or two
+    message = "Let me see if I can fix that for you! This might take a moment..."
+    await gh.post(event.data["issue"]["comments_url"], {}, data={"body": message})
+
+    # Get the username of the original committer
+    user = pr["user"]["login"]
+    email = "%s@users.noreply.github.com" % user
+
+    # Get the branch and pull request url
+    branch = pr["head"]["ref"]
+    clone_url = pr["head"]["repo"]["svn_url"]
+    naked_url = clone_url.replace("https://", "", 1)
+
+    # If for some reason the message below isn't generated?
+    message = "There was a problem running fixes! @spackbot needs a maintainer's help!"
```

Comment:
I thought about putting an entire try block around the with tmpdir, and then returning this message if that fails, but I think it would be more useful to have the actual error messages show up in logs. But this means we need someone that can easily look at the logs if something doesn't work. What would happen now if something unexpected happens is that spackbot just wouldn't respond, and that's probably not great. So maybe we would want here:

````python
try:
    <entire block>
except:
    message = "There was an error... get help, etc."
```
But then we'd need someone that can reliably look at the server logs, because it indicates some case we didn't expect. I'm sort of in favor of making the front end interaction clear, so I'm leaning towards having this try/except (although it's not added here). Let me know your thoughts. Right now this message is not doing anything.